plugins/nlp/manifest.cpp

```diff
@@ -0,0 +1,174 @@
+                catch (...)
+                {
+                    DBGLOG("Failed to load %s", dllname.str());
+                }
+            }
+        }
+
+        if (!manifest)
+        {
+            DBGLOG("Failed to initialize nlp manifest");
+            manifest.setown(createPTree());
+        }
+    }
+
+public:
+    Manifest(const Manifest &) = delete;
+    Manifest &operator=(const Manifest &) = delete;
+    ~Manifest()
+    {
+        DBGLOG("MANIFEST:  Destructor");
+    }
+
+    static std::shared_ptr<IManifest> getInstance(ICodeContext *ctx)
+    {
+        std::call_once(initFlag, [ctx]()
+                       { instance = std::shared_ptr<Manifest>(new Manifest(ctx)); });
+        return instance;
+    }
+
+    const char *stripLeadingCharacters(const char *input) const
+    {
+        while (*input == '.' || *input == '/' || *input == '\\')
+            ++input;
+        return input;
+    }
+
+    StringBuffer &makeResourcePath(const char *path, const char *basedir, StringBuffer &respath) const
+    {
+        StringBuffer abspath;
+        makeAbsolutePath(path, basedir, abspath);
+        return makePathUniversal(abspath, respath);
+    }
+
+    bool getResourceData(IPropertyTree *res, size32_t &len, const void *&data) const
+    {
+        if (res->hasProp("@id") && (res->hasProp("@header") || res->hasProp("@compressed")))
+        {
+            int id = res->getPropInt("@id");
+            return (dll->getResource(len, data, res->queryProp("@type"), (unsigned)id) && len > 0);
+        }
+        return false;
+    }
+
+    bool getResourceData(IPropertyTree *res, MemoryBuffer &content) const
+    {
+        size32_t len = 0;
+        const void *data = NULL;
+        if (getResourceData(res, len, data))
+        {
+            if (res->getPropBool("@compressed"))
+                decompressResource(len, data, content);
+            else
+                content.append(len, (const char *)data);
+            return true;
+        }
+        return false;
+    }
+
+    IPropertyTree *queryResourcePT(const char *_path) const
+    {
+        const char *path = stripLeadingCharacters(_path);
+        StringBuffer xpath;
+        if (!manifestDir.length())
+            xpath.setf("Resource[@filename='%s'][1]", path);
+        else
+        {
+            StringBuffer respath;
+            makeResourcePath(path, manifestDir.str(), respath);
+            xpath.setf("Resource[@resourcePath='%s'][1]", respath.str());
+        }
+
+        return manifest->queryPropTree(xpath.str());
+    }
+
+    bool ends_with(const std::string &str, const std::string &suffix) const
+    {
+        if (suffix.size() > str.size())
+            return false;
+        return std::equal(suffix.rbegin(), suffix.rend(), str.rbegin());
+    }
+
+    //  --- IManifest ---
+    virtual const char *extractResources(StringBuffer &sb) const
+    {
+        // dll->queryManifestFiles extracts the manifest files to a tmp folder the paths as a StringArray
```

Comment:
The IEngineContext version of this functionality looks easier to call, did you try using that?:

        IEngineContext *engine = codeCtx->queryEngineContext();
        if (engine)
        {
            StringArray manifestModules;
            engine->getManifestFiles("nlp", manifestModules);  //note use "UNKOWN" instead of "nlp" unless you add <Resource type="nlp" to your manifest

Reply:
dll->queryManifestFiles actually extracts the files to a tmp folder (which is neeed).